### PR TITLE
Feature/condo/sberdoma 311/user profile view

### PR DESCRIPTION
**Pull request by Dimitreee**

### CODE REVIEW

File: apps/condo/domains/user/components/NotDefinedField.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+interface INotDefinedFieldProps {
+    value?: string
+    render?: (value: string) => ReactElement
+}
+
+export const NotDefinedField: React.FC<INotDefinedFieldProps> = ({ value, render }) => {
+    const intl = useIntl()
+
+    if (!value) {
+        return intl.formatMessage({ id: 'errors.NotDefined' })
```
Comment (pahaz) [inline apps/condo/domains/user/components/NotDefinedField.tsx:14]: Why you don't want to extract Message const?

File: apps/condo/domains/user/components/UserOrganizationsList.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import React, { useCallback, useMemo } from 'react'
+import { NotDefinedField } from '@condo/domains/user/components/NotDefinedField'
+import { useQuery } from '@core/next/apollo'
+import { useIntl } from '@core/next/intl'
+import { useOrganization } from '@core/next/organization'
+import { Button } from '@condo/domains/common/components/Button'
+import { GET_ALL_EMPLOYEE_ORGANIZATIONS_QUERY } from '@condo/domains/organization/gql'
+import { useAuth } from '@core/next/auth'
+
+const OrganizationName = ({ name, organizationId, employeeOrganizationId, selectOrganization }) => {
```
Comment (pahaz) [inline apps/condo/domains/user/components/UserOrganizationsList.tsx:12]: Type?

File: packages/@core.next/auth.jsx [hunk 3]
```diff
@@ -25,6 +25,8 @@ const userFragment = `
   avatar {
     publicUrl
   }
+  phone
```
Comment (pahaz) [inline packages/@core.next/auth.jsx:28]: You don't have access to this fields for all users. Probably it should fail tests. 

You can create a custom GQL query for your profile. It's better to discuss it

File: apps/condo/pages/user/index.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    )
+}
+
+const HeaderAction = () => {
+    const intl = useIntl()
+
+    return (
+        <Space>
+            <Typography.Text style={{ fontSize: '12px' }}>
+                {intl.formatMessage({ id: 'Account' })}
```
Comment (pahaz) [inline apps/condo/pages/user/index.tsx:114]: Message ?

File: apps/condo/domains/user/components/UserMenu.tsx [hunk 5]
```diff
@@ -70,32 +98,27 @@ export const UserInfo = () => {
...
-                                        size='small'
-                                        style={{ backgroundColor: gold[4], borderRadius: '8px' }}
-                                        src={avatarUrl}
-                                        icon={!avatarUrl && <GitlabFilled style={{ color: colors.white }}/>}
-                                        alt={AvatarMessage}
-                                        className='avatar'
-                                    />
+                                    <AvatarContainer>
+                                        <UserAvatar/>
+                                    </AvatarContainer>
```
Comment (AntonAL) [inline apps/condo/domains/user/components/UserMenu.tsx:110]: Add a link to current user profile

Comment (AntonAL) [reply apps/condo/domains/user/components/UserMenu.tsx:110]: Add, please, a link also to current user name.
Now it's impossible to open current user profile page.

File: apps/condo/pages/user/index.tsx [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+                    <OrganizationRequired>
+                        <Row gutter={[0, 40]}>
+                            <Col span={3}>
+                                <UserAvatar borderRadius={24}/>
+                            </Col>
+                            <Col span={20} push={1}>
+                                <Row gutter={[0, 60]}>
+                                    <Col span={24}>
+                                        <Row gutter={[0, 40]}>
+                                            <Col span={24}>
```
Comment (AntonAL) [inline apps/condo/pages/user/index.tsx:43]: Too much for `span`, having sibling columns)
`24` is max value

File: apps/condo/pages/user/update.tsx [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+                                                            {PasswordMessage}
+                                                        </Typography.Text>
+                                                    </Col>
+                                                    <Col span={19} push={2}>
+                                                        <NotDefinedField value='******'/>
+                                                    </Col>
+                                                </Row>
+                                            </Col>
+                                            <Col span={24}>
+                                                <Link href={'/user/update'}>
```
Comment (AntonAL) [inline apps/condo/pages/user/update.tsx:81]: Link is to the same page

**Comment by AntonAL** [comment]
Please, test end-to-end what you are trying to pull ;)

File: packages/@core.next/auth.jsx (hunk 3 above)
Comment (Dimitreee) [reply packages/@core.next/auth.jsx:28]: https://doma.atlassian.net/browse/SBERDOMA-357

create task for this concern

**Review by dkoviazin — APPROVED** [review]
Ok

File: apps/condo/domains/user/components/UserPasswordResetButton.tsx [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+import { timer } from '../../common/utils/timer'
+import { START_PASSWORD_RECOVERY_MUTATION } from '../gql'
+import { extractRootDomain } from '@condo/domains/common/utils/url.utils.js'
+
+const PASSWORD_RESET_TIMEOUT = 60 // s
+
+const setCountDownDate = (date) => {
+    const domain = extractRootDomain(window.location.href)
+
+    document.cookie = `COUNTDOWN_DATE=${date}; path=/; domain=${domain}`
```
Comment (pahaz) [inline apps/condo/domains/user/components/UserPasswordResetButton.tsx:19]: it's better to use local storage. It's a front-end state and we don't need to know it on the server-side.

File: apps/condo/lang/ru.json [hunk 9]
```diff
@@ -276,12 +278,21 @@
...
   "ticket.EmptyList.header": "Список заявок пока пуст",
   "ticket.EmptyList.title": "Создайте первую, это просто",
   "TicketPurpose": "Назначение заявки",
   "ticketsTable.Number": "№",
   "ticketsTable.ResidentName": "Фио Жителя",
   "Title": "Заголовок",
+  "profile.Update": "Редактирование профиля",
+  "profile.ChangePasswordModalTitle": "Изменение пароля",
+  "profile.ChangePassword": "Сбросить пароль",
+  "profile.PasswordResetTimeout": "(сбросить ещё раз можно через 00:{countdown} секунд)",
```
Comment (pahaz) [inline apps/condo/lang/ru.json:291]:
```suggestion
  "profile.PasswordResetTimeout": "(сбросить ещё раз можно через 00: {countdown} секунд)",
```

**Review by pahaz — APPROVED** [review]

File: apps/condo/domains/user/components/UserPasswordResetButton.tsx (hunk 8 above)
Comment (Dimitreee) [reply apps/condo/domains/user/components/UserPasswordResetButton.tsx:19]: Optional now, i guess. This is will be good solution if we will want to sync data with other devices.

File: apps/condo/domains/user/components/UserPasswordResetButton.tsx [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+    return (
+        <Button
+            onClick={updatePassword}
+            type={'inlineLink'}
+            loading={passwordRecoveryLoading}
+            disabled={isCountDownActive}
+        >
+            {ChangePasswordLabel}
+            {isCountDownActive && (
+                <Typography.Text type={'secondary'}>
```
Comment (dkoviazin) [inline apps/condo/domains/user/components/UserPasswordResetButton.tsx:115]: The whole message flashes on countdown change.

**Review by dkoviazin — APPROVED** [review]
Looks cool

File: apps/condo/domains/user/components/UserPasswordResetButton.tsx (hunk 8 above)
Comment (pahaz) [reply apps/condo/domains/user/components/UserPasswordResetButton.tsx:19]: We also have a library for that

**Pull request merged**